pkg/commands/remote_read.go

```diff
@@ -0,0 +1,441 @@
+package commands
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/value"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/prometheus/prometheus/storage/remote"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/grafana/cortex-tools/pkg/backfill"
+)
+
+type RemoteReadCommand struct {
+	address        string
+	remoteReadPath string
+
+	tenantID string
+	apiKey   string
+
+	readTimeout time.Duration
+	tsdbPath    string
+
+	selector string
+	from     string
+	to       string
+}
+
+func (c *RemoteReadCommand) Register(app *kingpin.Application) {
+	remoteReadCmd := app.Command("remote-read", "TODO")
```

Comment:
I guess that `TODO` should be replaced before merging?

Reply:
Good catch :+1: